agent/src/Ingest.php

```diff
@@ -25,33 +25,33 @@ class Ingest
 
     private ?TimerInterface $flushBufferAfterDelayTimer = null;
 
+    private StreamBuffer|NullBuffer $buffer;
+
+    private StreamBuffer $streamBufferBackup;
+
     /**
      * @param  LoopInterface  $loop
      * @param  Browser  $browser
      * @param  (Closure(ResponseInterface $response, float $duration): mixed)  $onIngestSuccess
      * @param  (Closure(Throwable $e, float $duration): mixed)  $onIngestError
-     * @param  (Closure(float $duration): mixed)  $onExceededQuota
+     * @param  (Closure(float $duration): mixed)  $onOverQuota
      */
     public function __construct(
         private $loop,
         private $browser,
         private IngestDetailsRepository $ingestDetails,
-        private StreamBuffer $buffer,
+        StreamBuffer $buffer,
         private int $concurrentRequestLimit,
         private int $maxBufferDurationInSeconds,
         private Closure $onIngestSuccess,
         private Closure $onIngestError,
-        private Closure $onExceededQuota,
+        private Closure $onOverQuota,
     ) {
-        //
+        $this->buffer = $this->streamBufferBackup = $buffer;
     }
 
     public function write(string $payload): void
     {
-        if ($this->ingestDetails->quotaExceeded) {
-            return;
-        }
-
```

Comment:
Instead of doing this check on every write call, which is the most frequent activity for the agent, we now use a Null Object Pattern for the buffer.